src/app/ui/dialog/add-quest-dialog/add-quest-dialog.component.ts

```diff
@@ -0,0 +1,73 @@
+import { Component, OnInit, Inject } from '@angular/core';
+import { MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { PlayerQuestService } from '../../player-quest/player-quest.service';
+import { DocumentReference } from '@angular/fire/firestore';
+import { MatDialogRef } from '@angular/material/dialog';
+import { NotifyService } from '../../../core/notify.service';
+import { EmailService } from '../../../core/email.service';
+
+const CategoryList = [
+  'Core  - Infor Specific',
+  'Core  - Soft Skill',
+  'Product Specific Technical',
+  'Product Specific Functional',
+  'Certification ' 
+]
+
+@Component({
+  selector: 'add-quest-dialog',
+  templateUrl: './add-quest-dialog.component.html',
+  styleUrls: ['./add-quest-dialog.component.scss']
+})
+export class AddQuestDialogComponent implements OnInit {
```

Comment:
due to limitations in MS api, we cannot whitelist/register dynamic urls to access their API. Therefore it might be best to turn this into a page component and add a static route for it. I think it will also benefit us for future improvements in this area where we may need occupy more space that may not fit in the modal.